### PR TITLE
implement grouped gemm CI

### PR DESCRIPTION
**Pull request by aazz44ss**

### CODE REVIEW

**Review by jichangjichang — COMMENTED** [review]
Is that possible to verify our group gemm benchmark sizes?

**Comment by aazz44ss** [comment]
> Is that possible to verify our group gemm benchmark sizes?

Currently, program_option in hipblaslt doesn't support vector of arguments.

File: clients/benchmarks/client.cpp [hunk 1]
```diff
@@ -398,6 +398,10 @@ try
          bool_switch(&arg.scaleD_vector)->default_value(false),
          "Apply scaleD vector")
 
+        ("grouped_gemm",
```
Comment (jichangjichang) [inline clients/benchmarks/client.cpp:401]: Please update README.md of becnhmarks as well

File: clients/gtest/matmul_gtest.yaml [hunk 2]
```diff
@@ -563,4 +563,18 @@ Tests:
...
+  category: pre_checkin
+  function:
+    matmul: *hpa_half_precision
+  M: [127, 129]
+  N: [127, 129]
+  K: [127, 129]
+  transA_transB: *transA_transB_range
+  grouped_gemm: [1, 7]
+  alpha: 1
+  beta: 0
```
Comment (jichangjichang) [inline clients/gtest/matmul_gtest.yaml:577]: Suggest testing beta case as well: [ 0.0, 2.0 ]

Comment (aazz44ss) [reply clients/gtest/matmul_gtest.yaml]: done

File: clients/benchmarks/client.cpp (hunk 1 above)
Comment (aazz44ss) [reply clients/benchmarks/client.cpp:401]: done

**Review by jichangjichang — APPROVED** [review]
Approve this for now.
Please remember add these when API usage is locked down.
1. add group-gemm related test into CI test items "testing_matmul_bad_arg"
2. add the capability to test our exact bench sizes for CI.
3. add groupgemm API description into hipblaslt header for documentation

**Pull request merged**